### PR TITLE
More tests with some fixtures

### PR DESCRIPTION
**Pull request by david-doit-intl**
Signed-off-by: david-doit-intl <david@doit-intl.com>

### CODE REVIEW

**Comment by david-doit-intl** [comment]
Having trouble testing some of the more advanced stuff, I might need to invest in a mocking library

**Comment by david-doit-intl** [comment]
@stepanstipl this is in a passable state now

File: pkg/collector/cluster_test.go [hunk 1]
```diff
@@ -18,3 +18,18 @@ func TestNewClusterCollectorBadPath(t *testing.T) {
 		}
 	}
 }
+
+func TestNewClusterCollectorValidEmptyCollector(t *testing.T) {
+	testOpts := ClusterOpts{Kubeconfig: "../../fixtures/kube.config"}
+	collector, err := NewClusterCollector(&testOpts)
+
+	if err != nil {
+		t.Errorf("Should have crashed instead got un-parseable error: %s", err)
```
[the remark this answers is not in the document]
Comment (david-doit-intl) [reply pkg/collector/cluster_test.go:27]: yes whoops fixing!

File: fixtures/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: "extensions/v1beta1"
```
[the remark this answers is not in the document]
Comment (david-doit-intl) [reply fixtures/deployment.yaml:1]: Yep, sorry legacy, removed!

**Comment by david-doit-intl** [comment]
Done and done :)

**Comment by stepanstipl** [comment]
Cool, merged, thank you @david-doit-intl!

**Pull request merged**